lib/bolt/shell/powershell/snippets.rb

```diff
@@ -102,151 +102,11 @@ def shell_init
             "${boltBaseDir}\\hiera\\lib;" +
             $ENV:RUBYLIB
 
-            Add-Type -AssemblyName System.ServiceModel.Web, System.Runtime.Serialization
-            $utf8 = [System.Text.Encoding]::UTF8
-
-            function Write-Stream {
-            PARAM(
-              [Parameter(Position=0)] $stream,
-              [Parameter(ValueFromPipeline=$true)] $string
-            )
-            PROCESS {
-              $bytes = $utf8.GetBytes($string)
-              $stream.Write( $bytes, 0, $bytes.Length )
-            }
-            }
-
-            function Convert-JsonToXml {
```

Comment:
Were these functions just never used to begin with? (Just out of curiosity)

Reply:
They were a custom written set of functions to convert PSObjects to JSON text to send back to the ruby side of bolt if the target was powershell 2. Removing powershell 2 means we can remove all of this custom code and rely just on `ConvertTo-Json`